src/icon-button.tsx

```diff
@@ -356,4 +369,52 @@ export const BaseIconButton = (props: IconButtonProps): JSX.Element => {
     );
 };
 
-export default IconButton;
+type ToggleStateProps = {
+    Icon: React.FC<IconProps>;
+    type?: IconButtonType;
+    backgroundType?: IconButtonBackgroundType;
+} & AriaProps;
+
+interface BaseToggleProps {
+    checkedProps: ToggleStateProps;
+    uncheckedProps: ToggleStateProps;
+    onChange?: (checked: boolean) => void | undefined | Promise<void>;
+    checked?: boolean;
+    defaultChecked?: boolean;
+}
+
+export type ToggleIconButtonProps = BaseProps & BaseToggleProps;
+
+export const InternalToggleIconButton = React.forwardRef<
+    TouchableElement,
+    ToggleIconButtonProps & {isOverMedia?: boolean}
+>(({checked, defaultChecked, checkedProps, uncheckedProps, onChange, ...props}, ref) => {
+    const [checkedState, setCheckedState] = React.useState(!!defaultChecked);
+
+    const handleChange = () => {
+        if (checked === undefined) {
+            // if onChange is asynchronous, wait until it finishes and change the state if there was no error
+            const result = onChange?.(!checkedState);
+            if (result) {
+                return result.then(() => setCheckedState(!checkedState));
```

Comment:
I'd handle the rejection and do nothing (to me rejection would mean that the action was cancelled)

Reply:
Agree. Updated